src/hooks/useCategories.js

```diff
@@ -0,0 +1,49 @@
+import { addDoc, collection, onSnapshot } from "firebase/firestore";
+import { useEffect, useState } from 'react'
+import { db } from "../Firebase";
+
+const useCategories = () => {
+  const [categories, setCategories] = useState([]);
+  const [newCategory, setNewCategory] = useState("");
+  useEffect(() => {
+    const collectionCategory = collection(db, "category"); //日付を降順で並び替え
+
+    const unsubscribe = onSnapshot(collectionCategory, (snapshot) => {
+      const categoryData = snapshot.docs.map(doc => ({
+        id: doc.id,
+        ...doc.data()
+      }));
+      setCategories(categoryData)
+    });
+    return () => unsubscribe();
+  }, []);
+
+  const addCategory = async (isIncome, onClose) => {
+    if (!newCategory) return;
+    if (
+      newCategory === ""
+    ) {
+      alert("入力してください"); return
+    }
+    try {
+      await addDoc(collection(db, "category"), {
```

Comment:
`collection(db, "category")`は下でも使っているから useEffect の外側で定義してまとめてしまってもいいかもしれない

```suggestion
  const [newCategory, setNewCategory] = useState("");
  const collectionCategory = collection(db, "category"); //日付を降順で並び替え
  useEffect(() => {
    

    const unsubscribe = onSnapshot(collectionCategory, (snapshot) => {
      const categoryData = snapshot.docs.map(doc => ({
        id: doc.id,
        ...doc.data()
      }));
      setCategories(categoryData)
    });
    return () => unsubscribe();
  }, []);

  const addCategory = async (isIncome, onClose) => {
    if (!newCategory) return;
    if (
      newCategory === ""
    ) {
      alert("入力してください"); return
    }
    try {
      await addDoc(collectionCategory, {
```